### ISSUE TITLE
Question/Enhancement Changeable icon size in plot window

### ISSUE THREAD

**Issue by abuntrock** [OpenModelica/OpenModelica#11677]
### Description

I am not aware of a way to change the size of icons in a plot window (grid, Logarithmic axis, Plot setup, ...). If there is one please let me know. This can be an inconvenience if the user has many plots open besides each other and the plot window icons take up much space of the (smaller) plot window. 

### Steps to Reproduce

Change the toolbar icon size (Tools->Options->General) from 24 to 16 and simulate any model (e.g. Modelica.Blocks.Examples.PID_Controller). Go to the `Plotting ` pane and compare the toolbar icon size and the plot window icon size as shown in the attached screenshot. 

### Expected Behavior

There is a setting option (presumably under Tools->Options->Plotting) where the user can change the plot window icon size similar to the setting for the toolbar icon size under Tools->Options->General.

### Screenshots

My collegue @niklwors will add the screenshot as a comment.

### Version and OS

 - OpenModelica Version: 1.22.0
 - OS: Windows 10, 64 bit
 - MSL 4.0

Thanks for considering this.
Aaron Buntrock - Bosch Rexroth

**Comment by niklwors** [comment]
picture:
![image001](https://github.com/OpenModelica/OpenModelica/assets/12269776/4bccd7ca-af88-4082-a873-3e02505f697b)

**Comment by abuntrock** [comment]
Another remark regarding this topic. 
If you tackle this maybe you could add an option to hide the plot window icon toolbar entirely to free up even more space to inspect multiple plots in parallel.

In this OMEdit tutorial from 2019 one can see that this plot toolbar was much smaller in the past: https://vishia.org/mdlc/html/FB_Clang.en.html#chapter_8

The issue would be less important from our perspective if one could undock plot windows (#11785) and move them to other screens.

**Comment by ceraolo** [comment]
We already reduced the horizontal space a lot in comparison with last year's.

Maybe a further step is to substitute the Grid menu and Log x and Log y checkboxes, with a menu with checkable items. The menu may contain, for instance, the following items:

Simple grid
Detailed grid
Log X 
Log Y

(Simple grid would automatically unselect Detailed grid, and vice versa)



This will allow very small plots to be obtained, without hiding toold on the toolbar.

**Comment by casella** [comment]
> The issue would be less important from our perspective if one could undock plot windows (#11785) and move them to other screens.

This seems like a good idea in any case. Would you mind opening an enhancement ticket for that?

**Comment by adeas31** [comment]
Undocking plot windows is out of question. The are part of center widget.

**Comment by ceraolo** [comment]
A much better idea than that in https://github.com/OpenModelica/OpenModelica/issues/11677#issuecomment-2137760728:  we already have a Setup button. It is just a matter of adding there gird and log scale controls!

That would totally solve this ticket in a smooth and easy way.

**Comment by casella** [comment]
> Undocking plot windows is out of question. The are part of center widget.

Too bad. Any other way to spread plot windows on multiple screens? This could be quite useful.

**Comment by abuntrock** [comment]
> > The issue would be less important from our perspective if one could undock plot windows (#11785) and move them to other screens.
> 
> This seems like a good idea in any case. Would you mind opening an enhancement ticket for that?

@casella #11785 should be the enhacement ticket regarding plots on different screens (undocking) you need right?

**Comment by ceraolo** [comment]
> @ceraolo Adding the "Clear Plot Window" button to the individual plot window sounds like a good idea for us too.
good.
> 
> Your suggestion of moving grid and log controls inside the setup button would only reduce the horizontal size of the plot window toolbar right?
> 
> However, our point of this ticket and #11785 is that the user should be able to inspect 6 or 8 plots in parallel comfortably. Since I can already make the horizontal size of a plot window smaller than the width of all the plot window icons combined, I do not think that this change would help our usecase that much.

I also like to have several plots simultaneously. 
Consider the below plots. If we eliminate Grid, logY and logY, the minimum horizontal plot size will be comparable with the right plot below (think of substituting the "Grid" control with the three remaining buttons), which has become hardly readable for other reasons. Even if (as I often recommend to OM developpers) the number of numerical labels is automatically diminished for reduced width plots at least two labels must remain, so smaller plots than the right plot in the screen below, will be rare. 

Even in those rare cases, the tool buttons will be available clicking on ">>"  in the right part of the toolbar.

![image](https://github.com/OpenModelica/OpenModelica/assets/7373030/cd1850bc-80c9-4cf4-9cf6-de17a7118205)

So, I'm pretty sure that my proposal will never limit your choices.

I've written over the years a plotting program having in mind, among other things, the need to allow small plots, but not smaller than feasible. You can check, for instance, my result in the screen below (they are five independent windows, the green is my desktop).
The plot windows here have roughtly the minimum allowed width that would result if Grid, Log X and Log Y are removed.

![image](https://github.com/OpenModelica/OpenModelica/assets/7373030/68437842-348c-468e-8105-36b90d34cb57)

**Comment by adeas31** [comment]
#12857 fixes the actual problem reported in this issue.

It is not possible to undock plot windows.

@ceraolo I can possibly move `Clear Plot Window` and `Export CSV` button to plot window toolbar but this require quite many changes. I wonder if this is worth doing.
Regarding the horizontal space, the toolbar icons are replaced with `>>` so not sure if we really need to move grid and log buttons inside setup.

**Comment by abuntrock** [comment]
From my point of view we do not need grid and log buttons inside setup because of `>>` as you mentioned.

**Comment by ceraolo** [comment]
Thank you, @adeas31 . WIll check tomorrow

**Comment by ceraolo** [comment]
> 
> @ceraolo I can possibly move `Clear Plot Window` and `Export CSV` button to plot window toolbar but this require quite many changes. I wonder if this is worth doing. 

Ok. It's always a matter of cost/benefit ratio. If this requires substantial, I agree it can be disregarded.

>Regarding the horizontal space, the toolbar icons are replaced with `>>` so not sure if we really need to move grid and log buttons inside setup.

Agree. Now it is ok, for me.  Minimum plot size is not significantly limited from the toolbar size anymore.
It is often limited too much by the plot itself (probably related to the number of numerical labels on the axes), but this is another story and there are specific tickets for this.
